cmd/gobike-dataset/main.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/kevinburke/gobike"
+	"github.com/kevinburke/gobike/geo"
+)
+
+func run() error {
+	var format = flag.String("format", "csv", "CSV format")
+	flag.Parse()
+
+	trips, err := gobike.LoadDir(flag.Arg(0))
+	if err != nil {
+		return err
+	}
+	sort.Slice(trips, func(i, j int) bool {
+		return trips[i].StartTime.Before(trips[j].StartTime)
+	})
+
+	fw, err := os.Create(flag.Arg(1))
+	if err != nil {
+		return err
+	}
+	defer fw.Close()
+
+	w := csv.NewWriter(fw)
+
+	geolocate := func(lat, long float64) string {
+		if geo.SF.ContainsPoint(lat, long) {
+			return "San Francisco"
+		}
+		if geo.Oakland.ContainsPoint(lat, long) {
+			return "Oakland"
+		}
+		if geo.SanJose.ContainsPoint(lat, long) {
+			return "San Jose"
+		}
+		return ""
+	}
+
+	switch *format {
+	case "postgres":
+	default:
+		w.Write([]string{
+			"duration",
+			"start_time",
+			"end_time",
+			"start_station_id",
+			"start_station_name",
+			"start_station_latitude",
+			"start_station_longitude",
+			"start_station_city",
+			"end_station_id",
+			"end_station_name",
+			"end_station_latitude",
+			"end_station_longitude",
+			"end_station_city",
+			"bike_id",
+			"user_type",
+			"member_birth_year",
+			"member_gender",
+			"bike_share_for_all",
+		})
+	}
+
+	for i, trip := range trips {
+		if trip.EndTime.Before(trip.StartTime) {
+			// WTF?
```

Comment:
Should we panic here so we can identify the bad rows

Reply:
We probably want to clean it up when we're parsing the data itself.